test/addons-napi/test_buffer/test_buffer.c

```diff
@@ -0,0 +1,156 @@
+#include <stdlib.h>
+#include <string.h>
+#include <node_api.h>
+
+#define JS_ASSERT(env, assertion, message)              \
+  if (!(assertion)) {                                   \
+    napi_throw_error(                                   \
+        (env),                                          \
+        "assertion (" #assertion ") failed: " message); \
+    return;                                             \
+  }
+
+#define NAPI_CALL(env, theCall)                                           \
+  if (theCall != napi_ok) {                                               \
+    const char *errorMessage = napi_get_last_error_info()->error_message; \
+    errorMessage = errorMessage ? errorMessage : "empty error message";   \
+    napi_throw_error((env), errorMessage);                                \
+    return;                                                               \
+  }
+
+static const char theText[] =
+    "Lorem ipsum dolor sit amet, consectetur adipiscing elit.";
+
+static int deleterCallCount = 0;
+static void deleteTheText(void *data, void* finalize_hint) {
+  (void)finalize_hint;
+  free(data);
+  deleterCallCount++;
+}
+
+static void noopDeleter(void *data, void* finalize_hint) {
+  (void)finalize_hint;
+  deleterCallCount++;
+}
+
+void newBuffer(napi_env env, napi_callback_info info) {
+  napi_value theBuffer;
+  char *theCopy;
+  const unsigned int kBufferSize = sizeof(theText);
+
+  NAPI_CALL(env,
+            napi_create_buffer(
+                env,
+                sizeof(theText),
+                (void **)(&theCopy),
```

Comment:
We also try to avoid C-style casts, btw – if a proper cast pushes the line over 80 characters I personally would be okay, but this really doesn’t look like it does